tf2_ros/src/tf2_ros/buffer_client.py

```diff
@@ -171,13 +174,16 @@ def __is_done(self, state):
 
     def __process_goal(self, goal):
         self.client.send_goal(goal)
-        r = rospy.Rate(self.check_frequency)
+        # TODO(vinnamkim): rclpy.Rate is not ready 
+        # See https://github.com/ros2/rclpy/issues/186
+        #r = rospy.Rate(self.check_frequency)
         timed_out = False
-        start_time = rospy.Time.now()
+        clock = Clock()
+        start_time = clock.now()
         while not rospy.is_shutdown() and not self.__is_done(self.client.get_state()) and not timed_out:
-            if rospy.Time.now() > start_time + goal.timeout + self.timeout_padding:
+            if clock.now() > start_time + goal.timeout + self.timeout_padding:
                 timed_out = True
-            r.sleep()
+            sleep(self.check_frequency / 1000.0)
```

Comment:
Checking at 1000 times the desired check frequency seems overkill. Since it's a specific argument we should respect that.

This should also be capped using things like the goal timeout so we don't wait significantly past the resultant timeout if the check_frequency is low. Also please check the units it doesn't look right to sleep for a frequency, usually it should be 1/frequency.

Reply:
Fixed